### PR TITLE
expression: fix panic while call `Column.VecEvalInt`

### PR DESCRIPTION
**Pull request by lonng**
Signed-off-by: Lonng <heng@lonng.org>

### What problem does this PR solve? 

expression: fix panic while call `Column.VecEvalInt`

### What is changed and how it works?

The `Column.VecEvalInt` treats the `chunk.Chunk` as an `int64` container, but the original `chunk.Chunk` may have a different field type.

### Check List 

Tests 

 - Unit test
- Integration test https://github.com/tikv/copr-test/pull/22

Release note.
 - No need.

### CODE REVIEW

File: expression/column_test.go [hunk 1]
```diff
@@ -172,7 +172,7 @@ func (s *testEvaluatorSuite) TestColHybird(c *C) {
 		c.Assert(err, IsNil)
 		input.AppendBytes(0, num)
 	}
-	result, err := newBuffer(types.ETInt, 1024)
+	result, err := newBuffer(types.ETString, 1024)
```
Comment (SunRunAway) [inline expression/column_test.go]: Could you add an annotation to explain why use `VecEvalInt` but type is `ETString` here?

Comment (SunRunAway) [reply expression/column_test.go]: Maybe we should keep this regular test and add a new specific case.

Comment (lonng) [reply expression/column_test.go]: The case can be triggered if the result `chunk.Chunk` has a different field type with `Int`.

Comment (SunRunAway) [reply expression/column_test.go]: I get it, but it is not intuitive for the first-time code reader.

Comment (lonng) [reply expression/column_test.go]: Comments addressed

**Pull request merged**